core/src/runtime/mod.rs

```diff
@@ -82,25 +83,44 @@ pub struct Runtime {
 
     pub emit_events: bool,
 
-    /// Report of instruction calls.
-    pub report: InstructionReport,
+    /// Report of the program execution.
+    pub report: ExecutionReport,
 
     /// Whether we should write to the report.
     pub should_report: bool,
 }
 
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
-pub struct InstructionReport {
-    pub instruction_counts: HashMap<Opcode, u32>,
-    pub syscall_counts: HashMap<SyscallCode, u32>,
+pub struct ExecutionReport {
+    pub instruction_counts: HashMap<Opcode, u64>,
+    pub syscall_counts: HashMap<SyscallCode, u64>,
 }
 
-impl InstructionReport {
-    pub fn total_instruction_count(&self) -> u32 {
+impl ExecutionReport {
+    pub fn total_instruction_count(&self) -> u64 {
         self.instruction_counts.values().sum()
     }
 }
 
+impl Display for ExecutionReport {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "Instruction Counts:\n")?;
+        for (opcode, count) in &self.instruction_counts {
+            write!(f, "  {}: {}\n", opcode, count)?;
+        }
+        write!(
+            f,
+            "Total Instructions: {}\n",
+            self.total_instruction_count()
+        )?;
+        write!(f, "Syscall Counts:\n")?;
+        for (syscall, count) in &self.syscall_counts {
```

Comment:
it'd be nice to sort this in alphabetical order so its consistent

Reply:
updated

Reply:
Instruction Counts:
  add: 583439
  and: 151701
  auipc: 19488
  beq: 26917
  bge: 237
  bgeu: 5917
  blt: 1141
  bltu: 43457
  bne: 51442
  divu: 6
  ecall: 2264
  jal: 5372
  jalr: 38749
  lb: 10723
  lbu: 57950
  lw: 237094
  mul: 4036
  mulhu: 1152
  or: 301944
  sb: 58448
  sll: 278698
  sltu: 16766
  srl: 293010
  sub: 12382
  sw: 312781
  xor: 242242
Total Instructions: 2757356
Syscall Counts:
  16: 8
  26: 8
  0: 1
  6: 1091
  5: 1091
  2: 65
Total Syscall Count: 2264

Reply:
the difference in lower vs upper case is annoying, is it fine if I re-write all the Opcode enums https://github.com/succinctlabs/sp1/blob/dbed499f7aadbacc28a8e6e91a8761c5e7481d7d/core/src/runtime/opcode.rs#L72 to uppercase?

Reply:
can you just do .to_uppercase()?

Reply:
for consistency, should we show 0 for the missing ones?